filters/auth/oidc_introspection.go

```diff
@@ -120,10 +123,23 @@ func (filter *oidcIntrospectionFilter) Request(ctx filters.FilterContext) {
 
 func (filter *oidcIntrospectionFilter) Response(filters.FilterContext) {}
 
+func gjsonThisModifier(json, arg string) string {
+	return gjson.Get(json, "[@this].#("+arg+")").Raw
+}
+
 func (filter *oidcIntrospectionFilter) validateClaimsQuery(reqPath string, gotToken map[string]interface{}) bool {
-	gjson.AddModifier("_", func(json, arg string) string {
-		return gjson.Get(json, "[@this].#("+arg+")").Raw
-	})
+
+	gjsonModifierMutex.RLock()
+	// method is not thread safe
+	modExists := gjson.ModifierExists("_", gjsonThisModifier)
+	gjsonModifierMutex.RUnlock()
+
+	if !modExists {
+		gjsonModifierMutex.Lock()
+		// method is not thread safe
+		gjson.AddModifier("_", gjsonThisModifier)
```

Comment:
Isn't the problem rather that we call AddModifier in every call to validateClaimsQuery? Would it work if we simply just moved gjson.AddModifier to the package init function. If yes, I'd rather prefer that approach over syncing.

(In theory, the ideal would be if the auth package could use its own independent instance of the syntax, because then there is no danger that another package, own or imported, would try to use the same modifier name "_". But I see how this is currently not possible with the gjson library.)

Reply:
Very good point - I think you are on something here!
Makes sense to move this part into the init section of the filter rather then the execution. Let me try that.

Agreed, the lib should actually support instancing in order to provide a clear boundary, I've opened an issue there in this regard

Reply:
@aryszka your suggestion applied :+1: